app/javascript/components/common/Editor/CodeEditor.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { useState } from 'react'
+
+export function CodeEditor({ onSubmit }: { onSubmit: (code: string) => void }) {
+  const [code, setCode] = useState('')
+
+  function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault()
+
+    onSubmit(code)
+  }
+
+  function handleChange(e: React.ChangeEvent<HTMLTextAreaElement>) {
+    setCode(e.target.value)
+  }
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <label htmlFor="code">Code</label>
+      <textarea onChange={handleChange} id="code"></textarea>
```

Comment:
The ID is relatively generic and there might be a small risk of a duplicate ID being used. Do we have a strategy to make sure ID's are unique? For example by using some sort of "namespacing"?

Reply:
Yeah - I don't want us using `id` in React at all really. Is there ever a reason to? If it's for testing, can we use `data-xyz` attributes instead pls?

Reply:
We need IDs for labels.

Reply:
Ah yeah. I always forget this. so yeah, they should follow the class name thing then. So this would be `c-editor-code-textarea`